swarm/src/behaviour.rs

```diff
@@ -253,3 +271,33 @@ pub enum NotifyHandler {
     All
 }
 
+/// The available conditions under which a new dialing attempt to
+/// a peer is initiated when requested by [`NetworkBehaviourAction::DialPeer`].
+#[derive(Debug, Copy, Clone)]
+pub enum DialPeerCondition {
```

Comment:
Should we mark this with `#[non_exhaustive]` perhaps? Client code will most likely not match on this anyway and judging by the TODO below we may want to add other cases.

Reply:
Sounds good to me!

Reply:
On second thought, I think precisely because client code is not expected to match on it, but only libp2p-swarm code, we would want the exhaustiveness check when introducing new variants. `#[non_exhaustive]` seems to me primarily useful if one does not want to break client code matching on an enum which is known to be "open-ended" in the potential number of variants (and whereby handling of the new variants is optional in the vast majority of cases). Does that make sense?

Reply:
Nevermind, I was under the mistaken impression `non_exhaustive` would also affect the crate in which the enum is defined. Given that this is not so, my previous comment is nonsensical.

Reply:
Resolved by https://github.com/libp2p/rust-libp2p/pull/1519/commits/5c016989ecf4fc275c9f23c5ad729095efa7d7ac.